src/exoticatechnologies/modifications/exotics/impl/ShipRepulsorSystem.kt

```diff
@@ -0,0 +1,586 @@
+                val CCW2 = stage2CCW[index+1]
+                Global
+                        .getCombatEngine()
+                        .spawnEmpArcVisual(
+                                CCW1,
+                                ship,
+                                CCW2,
+                                ship,
+                                6f,
+                                Color.BLUE.darker(),
+                                Color.WHITE.darker()
+                        )
+            }
+            for (index in 0 until stage2CW.size) {
+                val from = stage1CW[index]
+                val to = stage2CW[index]
+                Global
+                        .getCombatEngine()
+                        .spawnEmpArcVisual(
+                                from,
+                                ship,
+                                to,
+                                ship,
+                                6f,
+                                Color.BLUE.darker(),
+                                Color.WHITE.darker()
+                        )
+            }
+            // end of stage2
+
+            // stage3 will just do CCW arcs between stage2ccw and stage3ccw
+            val stage3distance = fullRange
+            val stage3dotsPair = generateDots(center, stage3distance)
+            val stage3CCW = stage3dotsPair.second + stage3dotsPair.first.asReversed()
+            for (index in 0 until stage3CCW.size) {
+                val from = stage2CCW[index]
+                val to = stage3CCW[index]
+                Global
+                        .getCombatEngine()
+                        .spawnEmpArcVisual(
+                                from,
+                                ship,
+                                to,
+                                ship,
+                                6f,
+                                Color.BLUE.darker(),
+                                Color.WHITE.darker()
+                        )
+            }
+        }
+
+        //TODO move to CircleUtils, make the numDots a parameter and make the angleDeg depend on 360/numDots
+        /**
+         * Generates a pair of lists, going from 0-360 degrees with 10-degree increments, clockwise.
+         * The left list is reversed so that it represents CCW rotation from 0 to 180
+         *
+         * @return a pair of lists, first one being the 0-180 "right" list, second one being the reversed 180-360 "left" list
+         *
+         * @param center the center from which dots should diverge
+         * @param distance how far from the center should the dots be
+         */
+        private fun generateDots(center: Vector2f, distance: Float): Pair<List<Vector2f>, List<Vector2f>> {
+            val leftDots = mutableListOf<Vector2f>()
+            val rightDots = mutableListOf<Vector2f>()
+            val numDots = 36
+            for (i in 0 until numDots) {
+//            for (i in 0 ..numDots) {
+                val angleDeg = i * 10f
```

Comment:
this magic `10f` should become something like
`val angleIncrease = 360 / numDots`